src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs

```diff
@@ -103,14 +103,24 @@ internal void SetRequestLine(string req)
             try
             {
                 _version = new Version(parts[2].Substring(5));
-                if (_version.Major < 1)
-                    throw new Exception();
             }
             catch
             {
                 _context.ErrorMessage = "Invalid request line (version).";
                 return;
             }
+
+            if (_version.Major < 1)
+            {
+                _context.ErrorMessage = "Invalid request line (version).";
+                return;
+            }
+            if (_version != HttpVersion.Version10 && _version != HttpVersion.Version11)
```

Comment:
@stephentoub you can bring this commit into your branch to test it:
https://github.com/hughbe/corefx/commit/aa3489c991f26e8ddb91b50c8ba265b7145163ac